### PR TITLE
fixed valid moves and added a second move_piece! method

### PR DESCRIPTION
**Pull request by mstasss**

### CODE REVIEW

File: board.rb [hunk 1]
```diff
@@ -65,13 +65,20 @@ def board
   end
 
   def move_piece(start_pos, end_pos)
+    raise 'thats not a valid position' if start_pos.nil? || valid_pos?(end_pos) == false
+    raise "that's not a valid move" unless self[*start_pos].valid_moves.include?(end_pos)
+      self[*end_pos] = self[*start_pos]
+      self[*start_pos] = @np
+      self[*end_pos].pos = end_pos
```
Comment (dischorde) [inline board.rb]: Lines 68 and 70-72 sure look similar to `move_piece!`

File: test.rb [hunk 2]
```diff
@@ -12,81 +12,10 @@
...
 
 # TESTS
-# testing = Board.new
-# testing.move_piece([1, 3], [2, 3])
-# testing.move_piece([1, 4], [2, 4])
+testing = Board.new
+testing.move_piece([1,0],[2,0])
+testing.board_printer
 
-# testing.move_piece([0, 3], [1, 3])
```
Comment (dischorde) [inline test.rb:19]: Not a fan of the tests you had in here?

File: board.rb (hunk 1 above)
Comment (dischorde) [inline board.rb]:
```suggestion
    raise "that's not a valid move" unless self[*start_pos].valid_moves.include?(end_pos)
    move_piece!(start_pos, end_pos)
```

File: pieces/piece.rb [hunk 3]
```diff
@@ -8,31 +8,28 @@ def initialize(color, board, pos)
...
   def move_into_check?(end_pos)
     temp_board = board.dup
-    temp_board.move_piece(self.pos, end_pos)
+    temp_board.move_piece!(self.pos, end_pos)
     temp_board.in_check?(self.color)
   end
 
   def valid_moves
+    possible_moves = self.moves
+    possible_moves.reject do |pos|
```
Comment (dischorde) [inline pieces/piece.rb]:
```suggestion
    moves.reject do |pos|
```

File: test.rb (hunk 2 above)
Comment (mstasss) [reply test.rb:19]: heh

**Review by dischorde — APPROVED** [review]

**Pull request merged**